### PR TITLE
Use --stdin-display-name as filename when reading from stdin

### PR DESCRIPTION
**Pull request by bryanforbes**
When reading from `stdin`, `flake8.checker.FileChecker.filename` is set to `'-'`. This means that `PyiAwareFlakesChecker` does not replace `FlakesChecker` and things like forward references will fail the linter. Passing the contents of a file to `stdin` is a common thing for vim plugins to do, so this allows `pyi` files to be properly linted in vim.

### CODE REVIEW

File: pyi.py [hunk 1]
```diff
@@ -115,11 +115,16 @@ def handleNodeDelete(self, node):
...
+            filename = self.options.stdin_display_name
+        else:
+            filename = self.filename
+
+        if filename.endswith('.pyi') and plugin['plugin'] == FlakesChecker:
             LOG.info(
                 'Replacing FlakesChecker with PyiAwareFlakesChecker while '
                 'checking %r',
-                self.filename,
+                filename
```
Comment (ambv) [inline pyi.py]: nit: why did you remove the trailing comma?

**Comment by ambv** [comment]
This looks great. Please add this to the change log (the version will be 19.10.0, we are using CalVer) and reintroduce the missing trailing comma and we're good to go. Thanks! ✨ 🍰 ✨

File: pyi.py (hunk 1 above)
Comment (bryanforbes) [reply pyi.py]: Habit from JavaScript. I'll add it back.

**Comment by bryanforbes** [comment]
Don't you mean that the version will be 18.10.0?

**Comment by bryanforbes** [comment]
I've updated the change log with an entry about `--stdin-display-name` when reading from `stdin` under version 18.10.0 (because I didn't think you were going to wait another year to release this)

**Comment by ambv** [comment]
LOL, of course 18.10.0. I just wish 2018 was over, don't you?

**Comment by ambv** [comment]
Thank you for the quick turnaround and a worthwhile bugfix.
